scripts/gen-react-bindings.js

```diff
@@ -51,11 +51,11 @@ const getReactFileContents = async (svelteFilePath) => {
 import type * as React from 'react';
 import SvelteToReact, { type ReactProps } from '${svelteReactWrapperRelativePath}';
 import ${componentName} from './${fileName}';
-import type { ${componentName}Events, ${componentName}Props } from './${fileName}';
-const Untyped = SvelteToReact('${COMPONENT_PREFIX}-${fileNameWithoutExtension}', ${componentName});
-export default function ${componentName}React${funcConstraints}(props: React.PropsWithChildren<ReactProps<${componentName}Props${propParams}, ${componentName}Events${propParams}>>) {
-    return Untyped(props)
-}
+import type { ${componentName}Events as SvelteEvents, ${componentName}Props as SvelteProps } from './${fileName}';
+
+export type ${componentName}Props${funcConstraints} = ReactProps<SvelteProps${propParams}, SvelteEvents${propParams}>;
+const ${componentName}React = SvelteToReact('${COMPONENT_PREFIX}-${fileNameWithoutExtension}', ${componentName}) as ${funcConstraints}(props: React.PropsWithChildren<${componentName}Props${propParams}>) => JSX.Element;
```

Comment:
You can't wrap a React.forwardedRef in a function (it breaks the forward) so I updated the generator to just add the type explicitly.